src/main/java/life/qbic/samplecleaner/App.java

```diff
@@ -0,0 +1,103 @@
+package life.qbic.samplecleaner;
+
+import life.qbic.samplecleaner.tracking.SampleLocation;
+import life.qbic.samplecleaner.tracking.SampleLocationRepository;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ApplicationContext;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@SpringBootApplication
+public class App implements CommandLineRunner {
+
+  final ApplicationContext applicationContext;
+
+  final SampleLocationRepository sampleLocationRepository;
+
+  static final Pattern sampleCodePattern = Pattern.compile("^Q.*");
+
+  private static final Logger LOG = org.apache.logging.log4j.LogManager.getLogger(App.class);
+
+  @Autowired
+  public App(
+      ApplicationContext applicationContext, SampleLocationRepository sampleLocationRepository) {
+    this.applicationContext = applicationContext;
+    this.sampleLocationRepository = sampleLocationRepository;
+  }
+
+  public static void main(String[] args) {
+    SpringApplication.run(App.class, args);
+  }
+
+  @Override
+  public void run(String... args) {
+    LOG.info("Parsing sample id whitelist ...");
+    String filePath = args[0];
+    try {
+      HashSet<String> whiteListedSamples = (HashSet<String>) parseWhiteList(filePath);
```

Comment:
Why the type conversion? the method should return a set
```suggestion
      Set<String> whiteListedSamples = (HashSet<String>) parseWhiteList(filePath);
```

Reply:
done. why do you prefer set here?

Reply:
Its the Interface and not the Type :)